### PR TITLE
[PATCH API-NEXT v11] Stub compress API implementation

### PR DESCRIPTION
**Pull request by lumag**
While I'm wokring on real compression API implementation, I'd like to add this set of fixes, stubs and a test set submitted by @1234sv

### CODE REVIEW

**Comment by lumag** [comment]
@muvarov It seems your rebases have broken the api-next:
```
threads.c: In function 'odph_parse_options':
threads.c:345:17: error: 'odph_helper_options_t {aka struct <anonymous>}' has no member named 'proc'
   helper_options.proc = 1;
                 ^
```

**Comment by muvarov** [comment]
@lumag fixed.

**Comment by lumag** [comment]
@muvarov it would help if you'd care what are you closing. This PR is not merged. Please reopen it.

**Comment by muvarov** [comment]
@lumag  I have 2 buttons in web interface "Close and comment" and "Comment" . It looks like I clicked to wrong one somehow

File: test/validation/api/comp/comp.c [hunk 1]
```diff
@@ -0,0 +1,558 @@
+/* Copyright (c) 2014, Linaro Limited
```
Comment (muvarov) [inline test/validation/api/comp/comp.c]: 2018

File: test/validation/api/comp/comp.c [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+	    !(capa.comp_algos.bit.deflate))
+		rc = -1;
+	if (comp_alg == ODP_COMP_ALG_ZLIB &&
+	    !(capa.comp_algos.bit.zlib))
+		rc = -1;
+	if (comp_alg == ODP_COMP_ALG_LZS &&
+	    !(capa.comp_algos.bit.lzs))
+		rc = -1;
+
+	CU_ASSERT(!rc);
```
Comment (muvarov) [inline test/validation/api/comp/comp.c]: it's better to put CU_ASSERT in the place where error is actually appers. Error log will have line number which is useful information here.

File: test/validation/api/comp/comp.c [hunk 3]
```diff
@@ -0,0 +1,558 @@
...
+	odp_comp_session_param_init(&ses_params);
+	ses_params.op = op;
+	ses_params.comp_algo = comp_alg;
+	ses_params.hash_algo = hash_alg;
+	ses_params.compl_queue = suite_context.queue;
+	ses_params.mode = suite_context.op_mode;
+
+	session = odp_comp_session_create(&ses_params);
+	CU_ASSERT_FATAL(session != ODP_COMP_SESSION_INVALID);
+	CU_ASSERT(odp_comp_session_to_u64(session) !=
```
Comment (muvarov) [inline test/validation/api/comp/comp.c]: as I remember CU_ASSERT will not work after CU_ASSERT_FATAL, which you have at previous line

File: platform/linux-generic/odp_comp.c [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+void odp_comp_session_param_init(odp_comp_session_param_t *param)
+{
+	memset(param, 0, sizeof(odp_comp_session_param_t));
+}
+
+odp_comp_session_t
+odp_comp_session_create(const odp_comp_session_param_t *params)
+{
+	(void) params;
```
Comment (muvarov) [inline platform/linux-generic/odp_comp.c]: here if prior param_init to zero is expected you can add some check to capture missing param_init.

File: test/validation/api/comp/comp.c (hunk 3 above)
Comment (lumag) [reply test/validation/api/comp/comp.c]: `CU_ASSERT` works perfectly after `CU_ASSERT_FATAL` if conditions are different.

File: platform/linux-generic/odp_comp.c (hunk 4 above)
Comment (lumag) [reply platform/linux-generic/odp_comp.c]: We've never done such things before. I see no reason to do it now.

File: test/validation/api/comp/comp.c (hunk 2 above)
Comment (lumag) [reply test/validation/api/comp/comp.c]: Please compare this with crypto and ipsec code. It all has the same idiom.

File: test/validation/api/comp/comp.c (hunk 1 above)
Comment (lumag) [reply test/validation/api/comp/comp.c]: fixed

**Comment by Bill-Fischofer-Linaro** [comment]
@lumag please rebase this PR when convenient.

**Comment by lumag** [comment]
@Bill-Fischofer-Linaro done

File: test/validation/api/comp/comp.c [hunk 5]
```diff
@@ -1,4 +1,4 @@
-/* Copyright (c) 2014, Linaro Limited
+/* Copyright (c) 2018, Linaro Limited
```
Comment (muvarov) [inline test/validation/api/comp/comp.c]: @lumag  fix is  is merged to wrong patch

**Comment by Bill-Fischofer-Linaro** [comment]
@lumag Needs another rebase following v1.20 merge back to api-next. Thanks.

File: include/odp/api/spec/event.h [hunk 6]
```diff
@@ -78,6 +78,10 @@ extern "C" {
  *  List of event subtypes:
  * - ODP_EVENT_PACKET_BASIC
  *     - Packet event (odp_packet_t) with basic packet metadata
+ * - ODP_EVENT_PACKET_COMP
+ *     - Packet event (odp_packet_t) generated as a result of an compression/
```
Comment (Bill-Fischofer-Linaro) [inline include/odp/api/spec/event.h]: grammar: a compression

File: test/validation/api/comp/comp.c [hunk 7]
```diff
@@ -0,0 +1,589 @@
...
+		outdata = odp_packet_seg_data(pkt, seg);
+		outlen = odp_packet_seg_data_len(pkt, seg);
+		compare_len = outlen < (text_len - cmp_offset) ?
+			outlen : (text_len - cmp_offset);
+
+		CU_ASSERT(!memcmp(outdata,
+				  text + cmp_offset, compare_len));
+		cmp_offset += compare_len;
+		seg = odp_packet_next_seg(pkt, seg);
+	} while (seg != ODP_PACKET_SEG_INVALID || cmp_offset < text_len);
```
Comment (Bill-Fischofer-Linaro) [inline test/validation/api/comp/comp.c]: This doesn't seem right. If `seg == ODP_PACKET_SEG_INVALID` it doesn't matter if `cmp_offset < text_len`. You can't reference offsets into `ODP_PACKET_SEG_INVALID`.

File: test/validation/api/comp/test_vectors.h [hunk 8]
```diff
@@ -0,0 +1,1997 @@
+/* Copyright (c) 2014, Linaro Limited
```
Comment (Bill-Fischofer-Linaro) [inline test/validation/api/comp/test_vectors.h]: 2018 copyright, please.

**Comment by lumag** [comment]
This test suite now passes against implementation using miniz (which is not posted yet). I will post implementation after adding back digests support .

File: test/validation/api/comp/comp.c [hunk 9]
```diff
@@ -0,0 +1,584 @@
...
+		break;
+	default:
+		fprintf(stderr, "Unsupported compression algorithm\n");
+		return ODP_TEST_INACTIVE;
+	}
+
+	/* Hash algorithms */
+	switch (hash) {
+	case ODP_COMP_HASH_ALG_NONE:
+		break;
```
[the remark this answers is not in the document]
Comment (lumag) [reply test/validation/api/comp/comp.c]: yes

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**